licenses (#25181)
docker/docker.github.io: content/manuals/admin/organization/manage/manage-licenses.md

Reviewer: **though:** since the Docker Core license won't be available to all organizations at launch and will only be available to organizations with Docker Team or Docker Business + Docker AI Governance subscriptions, maybe we should add a callout to clarify this. Otherwise, customers without a Docker AI Governance subscription might start looking for Docker Core licenses and become confused why they can't find them in the Members section.

- To learn more about licenses, see [scale your subscription](/manuals/subscription/scale.md).

## Manage licenses

The **Members** page lets you track the number of available licenses for your organization and who currently holds a license. You can also assign or revoke licenses from this page. 

To manage licenses for your organization: 

1. Sign in to [Docker Home](https://app.docker.com), then choose your organization.
1. Select **Members** from the left navigation.
1. Select the action menu at the end of the row to assign or revoke an active license.
1. Optional. To manage automatic license assignment, enable or disable on a per-product basis with the **Automatically assign licenses** toggle. 

You must assign licenses manually, or configure automatic license assignment to consume a license. Inviting a new member to your organization may consume a seat for your Docker Team or Docker Business subscription, but won't auto-assign product licenses by default. Conversely, purchasing a set of licenses doesn't automatically assign licenses to existing members.

> ## Automated license assignment

Author: Change to automatic

Automated license assignment gives members a product license when they use a supported product for the first time.

> - For AI Governance and Docker Offload licenses, invoking `sbx` or signing into Docker Desktop (respectively) triggers an event that provisions licenses on a first come, first serve basis.

Reviewer 2: ```suggestion
- For AI Governance and Docker Core licenses, invoking `sbx` or signing into Docker Desktop (respectively) triggers an event that provisions licenses on a first come, first serve basis.
```

> - Licenses are assigned until exhausted. At that point, no more licenses will be assigned or added to your available license pool. 

Reviewer 2: ```suggestion
- Licenses are assigned until exhausted. Once the available licenses are exhausted, automatic license assignment will stop until more licenses are purchased or assigned licenses are revoked.
```

- Members may still use Docker Sandbox or Docker Desktop, but won't be held to the policies of those product licenses. 

Automated license assignment requires [setting up SSO](/manuals/enterprise/security/single-sign-on/connect.md), then [provisioning with SCIM or JIT](/manuals/enterprise/security/provisioning/_index.md). 

> [!NOTE]
> AI Governance licenses include SSO and provisioning features regardless of your Docker subscription status.  

> ## What’s next

Author: -_- i know

---
title: Manage license assignment
linkTitle: License assignment
description: Manage product licenses for your organization, including assignment, revocation, and automatic assignment.
keywords: licenses, organization, members, Docker Offload, AI governance, license assignment, admin console
weight: 30
---

> Licenses let you selectively choose which of your organization members have access to supported Docker products. Organization owners can oversee who on their team has active licenses, or configure licenses to assign automatically when members access supported Docker products. 

Author: @aevesdocker I forgot to add our note about licenses vs seats, but I imagine it front-loaded here. LMK what you think (this is also a note to myself for tomorrow)

Reviewer 3: Yes this is the bit that's missing for me on this page. I think it's super key to perhaps have a repeat paragraph that can be added to these pages which very clearly spell out the relationship between licence, subscription, and seat particularly as in the tip you say 'to learn more about licenses, see scale your subscription (so no matter where a user lands in docs, we're saying the same thing). I wonder if it's worth in this paragraph, also directly mentioning the Docker products we're talking about - Docker Core, DHI, Docker Sandboxes

> [!TIP]
> To learn more about licenses, see [scale your subscription](/manuals/subscription/scale.md), 
> or [contact sales](https://www.docker.com/pricing/contact-sales/) to purchase licenses.

## Manage licenses

The **Members** page lets you track the number of available licenses for your organization and who currently holds a license. You can also assign or revoke licenses from this page. 

To manage licenses for your organization: 

1. Sign in to [Docker Home](https://app.docker.com), then choose your organization.
1. Select **Members** from the left navigation.
1. Select the action menu at the end of the row to assign or revoke an active license.
1. Optional. To bulk assign or revoke licenses, choose the members you want to bulk manage, then select the **Bulk** menu. 

> 1. Optional. To manage automatic license assignment, turn off or turn on on a per-product basis with the **Automatically assign licenses** toggle. 

Author: ```suggestion
1. Optional. To manage automatic license assignment, turn off or turn on with the **Automatically assign licenses** toggle. 
```

You must assign licenses manually, or configure automatic license assignment to consume a license. Inviting a new member to your organization may consume a seat for your Docker Team or Docker Business subscription, but won't auto-assign product licenses by default. Conversely, purchasing a set of licenses doesn't automatically assign licenses to existing members.

## Automatic license assignment

Automatic license assignment gives members a product license when they use a supported product for the first time.

> - For AI Governance and Docker Core seats, invoking `sbx` or signing into Docker Desktop (respectively) triggers an event that provisions licenses on a first-come, first served basis.

Reviewer 3: ```suggestion
- For AI Governance and Docker Core, invoking `sbx` or signing into Docker Desktop (respectively) triggers an event that provisions licenses on a first-come, first served basis.
```
otherwise it conflates the two?

Reviewer 3: Also, is it AI Governance or Docker Sandboxes? What about Gordon?

Reviewer 3: I see further down that is is just Docker Sandboxes - I'd be explicit about that

Reviewer 4: > invoking `sbx`

I guess it's sbx sign-in that consumes a license, not the command invocation.

On AI Governance vs Docker Sandboxes - I would stick with AI Governance, since sbx without AI Governance is free to use / does not require a license. Only org-level policy enforcement does.

Author: When users use sbx, does it prompt them to sign in? What exactly are they signing into? 

(Just for clarity so we can correct my understanding: using sbx is tied to sign in, so if users 1-10 use sbx they get auto-assigned [if enabled], but user 11 can use sbx and not get assigned)

Reviewer 4: yes you are prompted to sign in - you sign into a Docker account.

I don't know what happens to user n+1 when n licenses have been consumed. That's a good thing to cover. Are they blocked? Do they get to continue using sbx with no governance? is the license allocated anyway?

They could always sign in with a non-corporate docker login, at least until sign-in enforcement is a thing (soon)

Author: They continue to use sbx as normal. This is covered in the manage-licenses "automatic assignment" section for MVP, and we're hoping to update this functionality soon after 🫡

- Licenses are assigned until exhausted. 
   - Once the available licenses are exhausted, automatic license assignment will stop until you purchase more licenses or revoke assigned licenses.

>    - Members can still use Docker Sandbox or Docker Desktop, but organization policies for those products won't affect their usage. 

Reviewer 3: This left me a bit confused, perhaps it needs to be a bit more stark like 'unlicensed users can still use X, but they're not covered by your security policies'? (have i understood that correctly?)

Also, with Desktop is this true? Can I still use DD if I'm not covered by a license? Is this the difference between a seat and a license? That I won't be covered by org security policies?

Automatic license assignment requires [setting up SSO](/manuals/enterprise/security/single-sign-on/connect.md), then [provisioning with SCIM or JIT](/manuals/enterprise/security/provisioning/_index.md). AI Governance licenses include SSO and provisioning features regardless of your Docker Core subscription. 

## What's next

> To learn more about Docker products, see:

Reviewer 3: I'd perhaps be tempted to direct users to learn more about subscriptions vs seat vs licenses rather than to other products at this point

---
title: Manage license assignment
linkTitle: License assignment
description: Manage product licenses for your organization, including assignment, revocation, and automatic assignment.

> keywords: licenses, organization, members, Docker Offload, AI governance, license assignment, admin console

Reviewer: nitpick: I know we don't have an official public name for DCP/DSoP, but maybe we should add `Docker Core` in addition to `Docker Offload` and `AI Governance` 🤷🏻‍♂️

weight: 30
---

Licenses let you selectively choose which of your organization members have access to supported Docker products. Organization owners can oversee who on their team has active licenses, or configure licenses to assign automatically when members access supported Docker products. Like Docker Core seats, licenses can be configured on a per member basis. 

> [!TIP]
> To learn more about product licenses, Docker Core seats, and other Docker add-ons see [scale your subscription](/manuals/subscription/scale.md), 
> or [contact sales](https://www.docker.com/pricing/contact-sales/) to purchase licenses.

## Manage licenses

The **Members** page lets you track the number of available licenses for your organization and who currently holds a license. You can also assign or revoke licenses from this page. 

To manage licenses for your organization: 

1. Sign in to [Docker Home](https://app.docker.com), then choose your organization.
1. Select **Members** from the left navigation.
1. Select the action menu at the end of the row to assign or revoke an active license.

> 1. Optional. To bulk assign or revoke licenses, choose the members you want to bulk manage, then select the **Bulk** menu. 

Reviewer: **suggestion:** the button name is `Bulk actions` if that's what you mean here.

<img width="514" height="200" alt="Image" src="https://github.com/user-attachments/assets/a1e1b039-d6c4-437b-82a7-2b0eb3b8af81" />

```suggestion
1. Optional. To bulk assign or revoke licenses, choose the members you want to bulk manage, then select the **Bulk actions** menu. 
```

1. Optional. To manage automatic license assignment, turn off or turn on with the **Automatically assign licenses** toggle. 

> You must assign licenses manually, or configure automatic license assignment to consume a license. Inviting a new member to your organization may consume a seat for your Docker Core subscription, but won't auto-assign product licenses by default. Conversely, purchasing a set of licenses won't trigger automatic assignment to existing members.

Reviewer: > Inviting a new member to your organization may consume a seat for your Docker Core subscription, but won't auto-assign product licenses by default.

**opinion:** This sounds a bit confusing because for the Docker Core product, 1x seat = 1x license.

Inviting a new member won't consume a seat/license unless the organization owner selects a license for a user they invite, which is optional. Product license will be consumed upon accepting an invitation and joining an organization.

If the organization owner sends an invite without selecting any licenses, and the user accepts it, then a user joins the organization and doesn't consume any seats/licenses.

## Automatic license assignment

> Automatic license assignment gives members a product license when they use a supported product for the first time. Automatic license assignment is available for AI Governance licenses. 

Reviewer: **suggestion:** since auto-assing via Docker Desktop will automatically provision Docker Core license, I would include it here too.

```suggestion
Automatic license assignment gives members a product license when they use a supported product for the first time. Automatic license assignment is available for Docker Core and AI Governance licenses. 
```

> - When you purchase AI Governance, signing into Docker Sandboxes with `sbx login` automatically provisions AI Governance licenses on a first-come, first served basis. 

Reviewer: **suggestion:** I would highlight that `sbx` is CLI, and add a reference to Docker Sanboxes docs.

```suggestion
- When you purchase AI Governance, signing into [Docker Sandboxes](https://docs.docker.com/ai/sandboxes/) with `login` command in `sbx` CLI (`sbx login`) automatically provisions AI Governance licenses on a first-come, first served basis. 
```

- Similarly, logins to Docker Desktop will automatically provision Docker Core licenses for AI Governance license-holding organizations that have available Docker Core seats.
- Licenses are assigned until exhausted. 
   - Once the available licenses are exhausted, automatic license assignment will stop until you purchase more licenses or revoke assigned licenses.
   - Members can still use Docker Sandbox or Docker Desktop, but organization policies for those products won't affect their usage. 

> AI Governance licenses include SSO and provisioning features regardless of your Docker Core subscription. Automatic license assignment requires [setting up SSO](/manuals/enterprise/security/single-sign-on/connect.md), then [provisioning with SCIM or JIT](/manuals/enterprise/security/provisioning/_index.md).

Reviewer: **suggestion:** for a reader who is unfamiliar with SSO/SCIM/JIT acronyms, it might not be obvious what those mean.

I would replace:

- SSO → [Single sign-on](https://docs.docker.com/enterprise/security/single-sign-on/) (SSO): 
- SCIM → [System for Cross-domain Identity Management](https://docs.docker.com/enterprise/security/provisioning/scim/provision-scim/) (SCIM)
- JIT → [Just-in-Time](https://docs.docker.com/enterprise/security/provisioning/just-in-time/) (JIT)

> AI Governance licenses include SSO and provisioning features regardless of your Docker Core subscription. Automatic license assignment requires [setting up SSO](/manuals/enterprise/security/single-sign-on/connect.md), then [provisioning with SCIM or JIT](/manuals/enterprise/security/provisioning/_index.md).

Author: ```suggestion
AI Governance licenses include single sign-on (SSO) and provisioning features regardless of your Docker Core subscription. Automatic license assignment requires [setting up SSO](/manuals/enterprise/security/single-sign-on/connect.md), then [provisioning with System for Cross-domain Identity Management (SCIM) or Just-in-Time (JIT)](/manuals/enterprise/security/provisioning/_index.md).
```
